src/main/java/queue/impl/ArrayQueueImpl.java

```diff
@@ -0,0 +1,98 @@
+package queue.impl;
+
+
+import java.lang.reflect.Field;
+import java.util.*;
+
+public class ArrayQueueImpl extends BaseQueue<Integer> {
+    private Integer[] elements = new Integer[10];
+    private int head = 0;
+    private int Cap = 10;
+    private int IndexCircularArray = 0;
+
+    private void ensureCapacity(int capacity) {
+        int firstArraySegment = size - head;
+        int boundary = 0;
+        if (capacity <= Cap && capacity != -1) {
+            return;
+        }
+        if (capacity == -1) {
+            Cap = Cap / 2;
+            firstArraySegment++;
+            boundary = 1;
+        } else {
+            Cap = Cap * 2;
+        }
+        Integer[] newElements = new Integer[Cap];
+        if (size >= 0) {
+            System.arraycopy(elements, head, newElements, 0, firstArraySegment);
+            System.arraycopy(elements, 0, newElements, firstArraySegment, head - boundary);
+        }
+        elements = newElements;
+        IndexCircularArray = size;
+        head = 0;
+    }
+
+    @Override
+    public Iterator<Integer> iterator() {
+        return new ArrayQueueIterator();
+    }
+
+    @Override
+    public boolean add(Integer integer) {
+        assert integer != null;
+        ensureCapacity(size + 1);
+        elements[IndexCircularArray] = integer;
+        IndexCircularArray = (IndexCircularArray + 1) % Cap;
+        size++;
+        return true;
+    }
+
+    @Override
+    public boolean offer(Integer integer) {
+        if (size < Cap) {
+            add(integer);
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public Integer poll() {
+        if (size == 0) {
+            return null;
+        }
+        size--;
+        int value = elements[head];
+        elements[head] = null;
+        head = (head + 1) % Cap;
+        if (size == Cap / 2 && size >= 10) {
```

Comment:
What if I ask you to increase size in 3 times? You will change each `2` literal?